pydocx/export/image_resize.py

```diff
@@ -0,0 +1,165 @@
+# coding: utf-8
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+    unicode_literals,
+)
+
+import base64
+from StringIO import StringIO
+from urlparse import urlparse
+import posixpath
+
+import requests
+
+from requests.exceptions import InvalidSchema
+
+from pydocx.util import uri
+from PIL import Image
+
+IMAGE_EXTENSIONS_TO_SKIP = ['emf', 'wmf', 'svg']
+IMAGE_FORMATS_TO_GIF_COMPRESS = ['BMP', 'TIFF']
+
+
+class ImageResizer(object):
+    def __init__(self, image_data, filename, x, y, skip_extensions=None):
+        self.image_data = image_data
+        self.filename = filename
+        self.x = self._get_dimension(x)
+        self.y = self._get_dimension(y)
+        self.image_format = None
+        self.image = None
+
+        self.skip_extensions = IMAGE_EXTENSIONS_TO_SKIP
+
+        if isinstance(skip_extensions, list):
+            self.skip_extensions += skip_extensions
+
+    def has_skipable_extension(self):
+        if not self.filename:
+            return False
+        lower_src = self.filename.lower()
+        extension = lower_src.rsplit('.')[-1]
+        return extension in self.skip_extensions
+
+    def has_height_and_width(self):
+        return bool(self.x and self.y)
+
+    def _get_dimension(self, dim):
+        if not dim:
+            return 0
+        try:
+            return int(dim.strip('px'))
+        except ValueError as e:
+            raise e
+
+    def init_image(self):
+        image_data = self.image_data
+        match = uri.is_encoded_image_uri(image_data)
+        if match:
+            image_data = base64.b64decode(match.group('image_data'))
+        try:
+            self.image = Image.open(StringIO(image_data))
+        except (IOError, SystemError) as e:
+            # PIL can't open it, return the image_data as is.
+            raise e
+
+    def resize_image(self):
+        # Let's not resize a base64 encoded image.
+        if uri.is_encoded_image_uri(self.image_data):
+            return
+        if not self.image:
+            return
+
+        _resized = False
```

Comment:
Well, in python an underscore to a variable mean that it's marked as private. It's not that "private" like we have for C++ or Java world, it's just a convention and is mostly used in class variables. This one at class scope, probably, and when I made available only to that function _ remains. Will remove it.